File: isu-sevice-maven/src/main/java/com/example/isusevicemaven/service/IsuService.java
package com.example.isusevicemaven.service;

import com.example.isusevicemaven.utils.StudyGroupApi;
import com.fasterxml.jackson.core.JsonProcessingException;
import entity.FormOfEducation;
import entity.StudyGroup;
import org.springframework.stereotype.Service;

@Service
public class IsuService {
    private final StudyGroupApi studyGroupApi;

    public IsuService(StudyGroupApi studyGroupApi) {
        this.studyGroupApi = studyGroupApi;
    }

    public StudyGroup expelAll(int groupId) throws JsonProcessingException {
        entity.StudyGroup studyGroup = studyGroupApi.studyGroupById(groupId);
        studyGroup.setStudentsCount(0L);
        return studyGroupApi.putStudyGroup(studyGroup);
    }

    public StudyGroup changeEduForm(int groupId, FormOfEducation formOfEducation) throws JsonProcessingException {
        entity.StudyGroup studyGroup = studyGroupApi.studyGroupById(groupId);
        studyGroup.setFormOfEducation(formOfEducation);
        return studyGroupApi.putStudyGroup(studyGroup);
    }
}

File: isu-service/src/main/kotlin/com/example/isuservice/service/IsuService.kt
package com.example.isuservice.service

import entity.FormOfEducation
import entity.StudyGroup
import org.springframework.http.HttpStatusCode
import org.springframework.stereotype.Service

@Service
class IsuService(
    val studyGroupApi: StudyGroupApi
) {
    fun expelAll(id: Int): Result {
        val studyGroup = when (val studyGroupById = studyGroupApi.studyGroupById(id)) {
            is Result.Error -> return studyGroupById
            is Result.Success -> studyGroupById.data
        }.apply {
            studentsCount = 0
            groupAdmin = null
        }
        return studyGroupApi.putStudyGroup(studyGroup)
    }

    fun changeEduForm(id: Int, newFormOfEducation: FormOfEducation): Result {
        val studyGroup = when (val studyGroupById = studyGroupApi.studyGroupById(id)) {
            is Result.Error -> return studyGroupById
            is Result.Success -> studyGroupById.data
        }.apply {
            formOfEducation = newFormOfEducation
        }
        return studyGroupApi.putStudyGroup(studyGroup)
    }

}

sealed interface Result {
    data class Success(val data: StudyGroup) : Result
    data class Error(val status: HttpStatusCode, val message: String?) : Result

}